packages/runtime/container-runtime/src/pendingStateManager.ts

```diff
@@ -80,7 +90,22 @@ export class PendingStateManager {
         return this.pendingMessagesCount !== 0;
     }
 
-    constructor(private readonly containerRuntime: ContainerRuntime) { }
+    public snapshot(): IPendingSnapshot | undefined {
```

Comment:
While "snapshot" seems good word to use here, it has connotation of something going to storage.
I'd try to find some other word to use, even though it may not seem like better choice.
serializeLocalState()?